src/main/java/bio/terra/controller/DataRepositoryServiceApiController.java

```diff
@@ -0,0 +1,134 @@
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Optional;
+
+@Controller
+public class DataRepositoryServiceApiController implements DataRepositoryServiceApi {
+
+    private Logger logger = LoggerFactory.getLogger(DataRepositoryServiceApiController.class);
+
+    private final ObjectMapper objectMapper;
+    private final HttpServletRequest request;
+    private final JobService jobService;
+    private final StudyRequestValidator studyRequestValidator;
+    private final StudyService studyService;
+    private final DatasetRequestValidator datasetRequestValidator;
+    private final DatasetService datasetService;
+    private final SamClientService samService;
+    private final IngestRequestValidator ingestRequestValidator;
+    private final FileService fileService;
+    private final DrsService drsService;
+
+    // needed for local testing w/o proxy
+    private final ApplicationConfiguration appConfig;
+
+    @Autowired
+    public DataRepositoryServiceApiController(
+            ObjectMapper objectMapper,
+            HttpServletRequest request,
+            JobService jobService,
+            StudyRequestValidator studyRequestValidator,
+            StudyService studyService,
+            DatasetRequestValidator datasetRequestValidator,
+            DatasetService datasetService,
+            SamClientService samService,
+            IngestRequestValidator ingestRequestValidator,
+            ApplicationConfiguration appConfig,
+            FileService fileService,
+            DrsService drsService
+    ) {
+        this.objectMapper = objectMapper;
+        this.request = request;
+        this.jobService = jobService;
+        this.studyRequestValidator = studyRequestValidator;
+        this.studyService = studyService;
+        this.datasetRequestValidator = datasetRequestValidator;
+        this.datasetService = datasetService;
+        this.samService = samService;
+        this.ingestRequestValidator = ingestRequestValidator;
+        this.appConfig = appConfig;
+        this.fileService = fileService;
+        this.drsService = drsService;
+    }
+
+/*
+    @InitBinder
+    protected void initBinder(final WebDataBinder binder) {
+        binder.addValidators();
+    }
+*/
+
+    @Override
+    public Optional<ObjectMapper> getObjectMapper() {
+        return Optional.ofNullable(objectMapper);
+    }
+
+    @Override
+    public Optional<HttpServletRequest> getRequest() {
+        return Optional.ofNullable(request);
+    }
+
+    private AuthenticatedUserRequest getAuthenticatedInfo() {
+        return AuthenticatedUserRequest.from(getRequest(), appConfig.getUserEmail());
+    }
+
+    @Override
+    public ResponseEntity<DRSAccessURL> getAccessURL(@PathVariable("object_id") String objectId,
+                                                     @PathVariable("access_id") String accessId) {
+        // TODO: this has to return DRSError - not an ErrorModel
+        // Have to figure out how to do that
```

Comment:
Not sure if this will work, but it would be nice if we can do something like declaring a more specific exception handler in this controller annotated with [@ExceptionHandler](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/ExceptionHandler.html) that can catch all the exceptions that come out of this controller (or at least an explicitly defined set of exceptions that we know should map to DRSError models).

Reply:
Since these use lower level components that throw our normal exceptions, I think these methods would have to catch all exceptions, wrap them in a special DRSError exception that included the http status code, and re-throw. The the @ExceptionHandler would unwrap it and generate the right response. Seems ugly, but I think it would work.